src/XMTP.types.ts

```diff
@@ -1,7 +1,24 @@
+import { ContentTypeID } from "./lib/ContentTypeID";
+
 export type ChangeEventPayload = {
   value: string;
 };
 
 export type XMTPViewProps = {
   name: string;
 };
+
+/*
+ *
+ * Represents encoded content and its metadata.
+ *
+ * @param {ContentTypeID} type - The content type ID for this content.
+ * @param {string} content - The encoded content data.
+ * @param {string} fallback - A fallback representation of the content, if any.
+ */
+
+export type EncodedContent = {
+  type: ContentTypeID;
```

Comment:
We'll still want to add the [`parameters`](https://github.com/xmtp/proto/blob/main/proto/message_contents/content.proto#L35) field from the EncodedContent protobuf definition since it's central to a few types of content types.
```suggestion
  type: ContentTypeID;
  parameters: { [key: string]: [value: string] }? ,
```